src/__swaps__/screens/Swap/types/assets.ts

```diff
@@ -0,0 +1,177 @@
+  balance: {
+    amount: string;
+    display: string;
+  };
+  native: {
+    balance: {
+      amount: string;
+      display: string;
+    };
+    price?: {
+      change: string;
+      amount: number;
+      display: string;
+    };
+  };
+}
+
+export type SearchAssetWithPrice = SearchAsset & ParsedAsset;
+export type ParsedSearchAsset = SearchAsset & ParsedUserAsset;
+
+export type ParsedAssetsDict = Record<UniqueId, ParsedUserAsset>;
+
+export type ParsedAssetsDictByChain = Record<ChainId | number, ParsedAssetsDict>;
+
+export interface ZerionAssetPrice {
+  value: number;
+  relative_change_24h?: number;
+}
+
+export type AssetApiResponse = {
+  asset_code: AddressOrEth;
+  decimals: number;
+  icon_url: string;
+  name: string;
+  chain_id?: number;
+  price?: {
+    value: number;
+    changed_at: number;
+    relative_change_24h: number;
+  };
+  symbol: string;
+  colors?: { primary?: string; fallback?: string; shadow?: string };
+  network?: ChainName;
+  networks?: {
+    [chainId in ChainId]?: {
+      address: chainId extends ChainId.mainnet ? AddressOrEth : Address;
+      decimals: number;
+    };
+  };
+  type?: AssetType;
+  interface?: 'erc-721' | 'erc-1155';
+};
+
+type AssetType = ProtocolType | 'nft';
+
+export interface ZerionAsset {
+  asset_code: AddressOrEth;
+  colors?: {
+    primary: string;
+    fallback: string;
+  };
+  implementations?: Record<string, { address: Address | null; decimals: number }>;
+  mainnet_address?: AddressOrEth;
+  name: string;
+  symbol: string;
+  decimals: number;
+  type?: AssetType;
+  icon_url?: string;
+  is_displayable?: boolean;
+  is_verified?: boolean;
+  price?: ZerionAssetPrice;
+  network?: ChainName;
+  bridging: {
+    bridgeable: boolean;
+    networks: { [id in ChainId]?: { bridgeable: boolean } };
+  };
+}
+
+// protocols https://github.com/rainbow-me/go-utils-lib/blob/master/pkg/enums/token_type.go#L44
+export type ProtocolType =
+  | 'aave-v2'
+  | 'balancer'
+  | 'curve'
+  | 'compound'
+  | 'compound-v3'
+  | 'maker'
+  | 'one-inch'
+  | 'piedao-pool'
+  | 'yearn'
+  | 'yearn-v2'
+  | 'uniswap-v2'
+  | 'aave-v3'
+  | 'harvest'
+  | 'lido'
+  | 'uniswap-v3'
+  | 'convex'
+  | 'convex-frax'
+  | 'pancake-swap'
+  | 'balancer-v2'
+  | 'frax'
+  | 'gmx'
+  | 'aura'
+  | 'pickle'
+  | 'yearn-v3'
+  | 'venus'
+  | 'sushiswap';
+
+export type AssetMetadata = {
+  circulatingSupply: number;
+  colors?: { primary: string; fallback?: string; shadow?: string };
+  decimals: number;
+  description: string;
+  fullyDilutedValuation: number;
+  iconUrl: string;
+  marketCap: number;
+  name: string;
+  networks?: {
+    [chainId in ChainId]?: {
+      address: chainId extends ChainId.mainnet ? AddressOrEth : Address;
+      decimals: number;
+    };
+  };
+  price: {
+    value: number;
+    relativeChange24h: number;
+  };
+  symbol: string;
+  totalSupply: number;
+  volume1d: number;
+};
+
+export type UniqueId = `${Address}_${ChainId}`;
```

Comment:
getting war flashbacks here, kind of a nbd if contained to swaps but testy about having too patterns here